### PR TITLE
possible fix for #3746

### LINKED ISSUE #3746: Failure to parse origin-spanning features
### Setup

I am reporting a problem with Biopython version, Python version, and operating
system as follows:

```
3.9.6 (default, Jun 29 2021, 06:20:32) 
[Clang 12.0.0 (clang-1200.0.32.29)]
CPython
macOS-10.15.7-x86_64-i386-64bit
1.79
```

### Expected vs. Actual behaviour

Loading the attached GenBank file (renamed for upload to GitHub): [pME_B_0_08_0005_Akan(Vn)_(sfgfp).gb.txt](https://github.com/biopython/biopython/files/7210096/pME_B_0_08_0005_Akan.Vn._.sfgfp.gb.txt)
discards the origin-spanning feature as being from an undefined sequence topology, despite the fact that sequence topology is recognized as circular.

### Steps to reproduce

````
> records = [r for r in SeqIO.parse('pME_B_0_08_0005_Akan(Vn)_(sfgfp).gb','gb')]
/usr/local/lib/python3.9/site-packages/Bio/GenBank/__init__.py:352: BiopythonParserWarning: It appears that '3093..570' is a feature that spans the origin, but the sequence topology is undefined. Skipping feature.
  warnings.warn(
> records[0].annotations
{'topology': 'circular', 'data_file_division': 'UNK', 'date': '22-SEP-2021', 'accessions': ['urn.local...ef-c2dp9c9'], 'keywords': [''], 'source': '', 'organism': '.', 'taxonomy': [], 'molecule_type': 'DNA'}
````

### PR DESCRIPTION
**Pull request by manulera**
- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit``
locally, and understand that continuous integration checks will be used to
confirm the Biopython unit tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Closes #3746

I encountered the same problem described by @jakebeal. Instead of passing the `seq_type` to the function `_loc`, one could pass a boolean indicating whether the sequence is circular. This can accessed from `self.data.annotations["topology"]`. This is correctly set for the two examples that I provided for the test:

* One was mentioned in #3746 by @starling13 and should return an error: trying to set an origin-spanning feature in a linear molecule.
* The other one is AddGene plasmid [39296](https://www.addgene.org/39296/) , a plasmid commonly used in many labs that gave the problem described by @jakebeal.

This is one of my first times contributing to Open Source, so please let me know if I should do something else.

### CODE REVIEW

File: Bio/GenBank/__init__.py [hunk 1]
```diff
@@ -1099,6 +1099,12 @@ def location(self, content):
 
         cur_feature = self._cur_feature
 
+        # Check if the sequence is circular for features that span the origin
+        seq_is_circular = (
+            "topology" in self.data.annotations
+            and self.data.annotations["topology"] == "circular"
+        )
+
```
Comment (peterjc) [inline Bio/GenBank/__init__.py:1107]: That can probably be shortened to:

```python
seq_is_circular = self.data.annotations.get("topology","") == "circular"
```

And perhaps ``is_circular`` is enough for the variable name?

Comment (peterjc) [reply Bio/GenBank/__init__.py:1107]: However, the changes appears to have lost the ``.lower()`` and use of ``in`` we had before (e.g. things like "Circular DNA" would have been accepted).

So maybe this?:

```python
seq_is_circular = "circular" in self.data.annotations.get("topology","").lower()
```

Comment (manulera) [reply Bio/GenBank/__init__.py:1104]: Hi @peterjc, thanks for the quick revision. I have added those changes in the last commit.

**Review by peterjc — APPROVED** [review]
Looks good to me (just waiting for tests to finish).

I think we can squash-and-merge this into one commit?

**Comment by manulera** [comment]
> Looks good to me (just waiting for tests to finish).
> 
> I think we can squash-and-merge this into one commit?

Hey @peterjc sure. Do I have to do the squash locally and push it, or can this be done directly from the pull request here in GitHub?

**Comment by peterjc** [comment]
The maintainers of the repository have a big green merge button, with a drop down things like squash-and-merge or rebase-and-merge. So you don't need to do anything for that :)

**Comment by peterjc** [comment]
Merged, thank you.

**Pull request merged**